cmd2/table_creator.py

```diff
@@ -0,0 +1,748 @@
+# coding=utf-8
+"""
+cmd2 table creation API
+This API is built upon two core classes: Column and TableCreator
+The general use case is to inherit from TableCreator to create a table class with custom formatting options.
+There are already implemented and ready-to-use examples of this below TableCreator's code.
+"""
```

Comment:
The code, example, and tests all look good and the test coverage is excellent.  This looks like a nice new feature.  

What is missing is documentation and a changelog note.  I also since this feature is brand  new and significant, I think we should mark this feature as being in an "alpha" or "beta" state and subject to change without it counting as a backwards-incompatible change which would trigger a major version update.  Perhaps we need to agree on good way to do this for features.  Previously we discussed ways to make a feature as deprecated but we haven't discussed ways of marking them as in-development.  @kotfu Do you have any suggestions in this regard?

I think the following should be added:
1. CHANGELOG entry for an enhancement describing this new feature which also flags it as in-development, beta, or alpha or something like that
2. Sphinx doc entry of some kind, probably feature and API - the feature one can be brief and link to the example
3. We should probably convert the existing table_display.py example to use this new built-in capability

Reply:
Agree with @tleonhardt, @kmvanbrunt has made a nice new feature here. I made a few minor comments, but this is a really good PR. Agree with the checklist above to complete before merging.

Re alpha and beta state: the only way I know to do that is to make a feature branch that we tag with releases like "cmd2-1.1b1". We keep master as is so we could release "cmd2-1.0.x" versions before "cmd2-1.1" is completed.

Reply:
I just removed the `tableformatter` example instead of converting it. The `TableCreator` example shows off our features pretty well.